Remove redundant KubeletPSI feature label from sig-node e2e test

goodhee opened the pull request:
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This PR removes a redundant feature label from the sig-node e2e test suite. 

The test in `summary_test.go` already checks the KubeletPSI feature gate directly using `utilfeature.DefaultFeatureGate.Enabled(features.KubeletPSI)`, making the `feature.KubeletPSI` label redundant.

This is part of the cleanup effort tracked in issue #134172 to remove unnecessary feature labels from sig-node e2e tests.

#### Which issue(s) this PR is related to:

Partial fix for #134172

#### Special notes for your reviewer:

This is part of the broader cleanup effort to review and clean up feature labels in sig-node e2e tests as outlined in issue #134172. The goal is to:

1. Remove feature labels that are just used for categorization
2. Replace feature labels with feature gates where appropriate  
3. Ensure feature labels that indicate special environments are well documented

This specific change removes a redundant feature label where the feature gate check already exists in the test code.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

Commit:
Replace feature label with WithFeatureGate for KubeletPSI test

- Remove feature.KubeletPSI label as it may be redundant for categorization
- Add framework.WithFeatureGate(features.KubeletPSI) for proper feature gate checking
- Remove redundant utilfeature.DefaultFeatureGate.Enabled check in BeforeEach

This addresses the cleanup goal in issue #134172 to replace feature labels
with feature gates where appropriate.

kannon92 commented:
> Hi @kannon92, I want to make sure I understood your feedback correctly.
> 
> Based on your comment "removing the psi feature is fine but we still need the feature gate label", I interpreted this as:
> 
> * Remove the `feature.KubeletPSI` label ❌
> * Add `framework.WithFeatureGate(features.KubeletPSI)` ✅
> * Remove redundant feature gate check in BeforeEach ✅
> 
> Could you confirm if this is the correct approach? Or did you mean to keep both the feature label AND add WithFeatureGate?
> 
> Thanks for the clarification!

Drop feature.KubeletPSI but add the WithFeatureGate.

kannon92 commented:
/ok-to-test

SergeyKanzhelev commented:
/assign

eun2ce commented:
The `pull-kubernetes-unit` test failure appears unrelated to this PR's changes.

This PR only modifies: `test/e2e_node/summary_test.go`
Failing test: `TestRoundTripTypes` in `pkg/api/testing/serialization_test.go`

Could a reviewer or SIG member help verify if this is a known issue?

SergeyKanzhelev commented on test/e2e_node/summary_test.go:
```diff
@@ -368,11 +367,8 @@ var _ = SIGDescribe("Summary API", framework.WithNodeConformance(), func() {
 	})
 
-	framework.Context("when querying /stats/summary under pressure", feature.KubeletPSI, framework.WithSerial(), func() {
+	framework.Context("when querying /stats/summary under pressure", framework.WithSerial(), framework.WithFeatureGate(features.KubeletPSI), func() {
```
cc @mariafromano-25

Commit:
Add WithNodeConformance to KubeletPSI test context

- Add framework.WithNodeConformance() to indicate test doesn't require special environment
- This addresses reviewer feedback on PR #134303
